sovrin_client/agent/endpoint.py

```diff
@@ -57,13 +70,26 @@ def transmitToClients(self, msg: Any, remoteNames: List[str]):
     # TODO: Rename method
     def baseMsgHandler(self, msg):
         logger.debug("Got {}".format(msg))
-        self.msgHandler(msg)
+        self._msgHandler(msg)
 
     def connectTo(self, ha):
-        remote = self.findInRemotesByHA(ha)
+        remote = self.stack.findInRemotesByHA(ha)
```

Comment:
All this logic is stack specific, its saying "if its not connected to an ha then connect", abstract this into stack.